### PR TITLE
feat: Add swap UI machine

### PR DESCRIPTION
**Pull request by cawabunga-bytes**
PR:
- adds a new state machine `swapUIMachine`, which will be responsible for entire swap flow

Follow-up will be incorporating swapMachine from swap-facade repository.

### CODE REVIEW

**Comment by jobotics** [comment]
Let's create a dedicated folder for all machines and their tests.

Path change:
From: `src/features/swap/machines/swapUIMachine.test.ts`
To: `src/features/machines/swapUIMachine.test.ts`

Here are two objective reasons:
- It will maintain a clear structure, as machines are part of the SDK's core features.
- If we decide to move all machines to a separate package, it will be easier to extract the entire folder.

File: src/features/swap/machines/swapUIMachine.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    },
+    isQuoteRelevant: () => {
+      throw new Error("not implemented")
+    },
+  },
+}).createMachine({
+  /** @xstate-layout N4IgpgJg5mDOIC5SwO4EMAOBaArgSwDpI8AXPAOygGJYcAjAW1IG0AGAXUVAwHtZS8PclxAAPRAEYJADgIB2AKwA2JdIkBmVqyUKFAJgA0IAJ6S9e+UoCcc6bfV7pe1noC+ro6ky5CxMpSoKDBwSNk4kEF5+MiERcQQsPQ0CVl1pKx1pFRdDE0REuQIdABY7BTslYpctNw8QL2x8IggBAKCQ5glw7j4BWIj4rAVtAgybCWsJPSUpOSNTBIUCCTkVCYU5c3T1HfdPdEbfFv8oAjwIABswKlFYEjQSMAI0ADNHgCcACgBHHB5HgAKPAuFwoUAAkuQPgA3NAXACUNAOPmarVO5yuYREUT6wgG+SULgI0gUO1Jq3SVisuQWE2Ky1UKyk0m0TiU6j29WRTT8YIIv3+YKoECETwo0J4AGsngLHgBFHBgd54JVYiI4mJ40CDdTJaTFKR6KwOBTFTSreaIPRyVgpazFcrFTZ6BRTJSchoo3mUfl-E5UJXvHjvAgYC4PF7Bhi+-5gBVKlXvNU9aKCLVifJTelVbZKRTFKypPOWhDW23aKwO-XO121fbeJq0Rikf0i8hi8gS6UET2N+hMEgnBDingAYweabCycivU1cUQuisRRUjipw1KMxLElYEgIpvKrAy6krxWKHu5hFHPAYYbAjxo93eJAA8tDVRxsbO0-OEKal7o5ELQC0isEkS3MQpWQPBQqQcHcJHcOpyB4CA4BEXs8E-VN+m1fInCWFRAMAlQqWKHQS0SIlNArdR1xJItzwbI40Sw3EfywOROOXIiMmsStyLyBIpHpGxOP1axtGmRxGMOVETjOS4wFYud8QSTYl0ImxeNIgTaSUCwyQLJ0Vk2TYzzqDC5L5WUwWU79VMSSo9y0Q8dnwyttC3J1LBkaRjxsHQMgUGSUSbAcTjsnCMwQdRFAIPQDSycxaKsfRihLY1ljI0opjkBxNA5CyLwIK8byuR5IvTeI5AkekZBUB0DQNPMaUkGR4tddQnHUUpSXtRDXCAA */
```
Comment (jobotics) [inline src/features/swap/machines/swapUIMachine.ts:45]: Why do we need this?

Comment (cawabunga-bytes) [reply src/features/swap/machines/swapUIMachine.ts:45]: It maintains nice layout for XState VSCode Extension

**Comment by cawabunga-bytes** [comment]
@jobotics done

**Review by jobotics — APPROVED** [review]

**Pull request merged**